src/sdk/config_fs.js

```diff
@@ -742,24 +764,43 @@ class ConfigFS {
 
     /**
      * create_bucket_config_file creates bucket config file
-     * @param {string} bucket_name
-     * @param {*} data
-     * @returns {Promise<void>} 
+     * @param {Object} bucket_data
+     * @returns {Promise<String>} 
      */
-    async create_bucket_config_file(bucket_name, data) {
-        const bucket_path = this.get_bucket_path_by_name(bucket_name);
-        await native_fs_utils.create_config_file(this.fs_context, this.buckets_dir_path, bucket_path, data);
+    async create_bucket_config_file(bucket_data) {
+        const { parsed_bucket_data, string_bucket_data } = this._prepare_for_bucket_schema(bucket_data);
+        const bucket_path = this.get_bucket_path_by_name(bucket_data.name);
+        await native_fs_utils.create_config_file(this.fs_context, this.buckets_dir_path, bucket_path, string_bucket_data);
+        return parsed_bucket_data;
     }
 
     /**
-     * update_bucket_config_file updates bucket config file
-     * @param {string} bucket_name
-     * @param {*} data
-     * @returns {Promise<void>} 
+     * _prepare_for_bucket_schema processes bucket data before writing it to the config dir and does the following -
+     * 1. removes API bucket properties
+     * 2. removes undefined properties, unwrap sensitive_strings and creation_data to string
+     * 3. checks bucket schema validation
```

Comment:
I would remove the "checks", for example:
```suggestion
     * 3. validate the bucket_data against the bucket schema
```